src/pemfile.rs

```diff
@@ -72,29 +30,57 @@ pub enum Error {
     Base64Decode(String),
 }
 
+#[cfg(feature = "std")]
+impl From<Error> for io::Error {
+    fn from(error: Error) -> Self {
+        match error {
+            Error::MissingSectionEnd { end_marker } => io::Error::new(
+                ErrorKind::InvalidData,
+                format!(
+                    "section end {:?} missing",
+                    String::from_utf8_lossy(&end_marker)
+                ),
+            ),
+
+            Error::IllegalSectionStart { line } => io::Error::new(
+                ErrorKind::InvalidData,
+                format!(
+                    "illegal section start: {:?}",
+                    String::from_utf8_lossy(&line)
+                ),
+            ),
+
+            Error::Base64Decode(err) => io::Error::new(ErrorKind::InvalidData, err),
+        }
+    }
+}
+
+impl From<parent::Error> for Error {
+    fn from(error: parent::Error) -> Self {
+        match error {
+            parent::Error::MissingSectionEnd { end_marker } => {
+                Error::MissingSectionEnd { end_marker }
+            }
+            parent::Error::IllegalSectionStart { line } => Error::IllegalSectionStart { line },
+            parent::Error::Base64Decode(str) => Error::Base64Decode(str),
+
+            // this is a necessary bodge to funnel any new errors into our existing type
+            // (to which we can add no new variants)
```

Comment:
maybe worth filing an issue with a "next release" label and citing it here?

Reply:
I was thinking we would probably not do a 3.x release for this crate?

Reply:
Would be good to avoid it, yeah.

Reply:
WFM